src/test/java/uk/gov/hmcts/reform/sscs/utils/SscsCaseDataUtilsTest.java

```diff
@@ -0,0 +1,381 @@
+        SscsCaseData sscsCaseData = Mockito.mock(SscsCaseData.class);
+        // when
+        Mockito.when(sscsCaseData.getOtherParties()).thenReturn(null);
+        //then
+        assertTrue(SscsCaseDataUtils.getParties(sscsCaseData).isEmpty());
+    }
+
+    @Test
+    void shouldGetHearingWindowWhenUrgentCase() {
+        // given
+        SscsCaseData sscsCaseData = Mockito.mock(SscsCaseData.class);
+        // when
+        Mockito.when(sscsCaseData.getEvents()).thenReturn(getEventsOfCaseData());
+        Mockito.when(sscsCaseData.getUrgentCase()).thenReturn((YesNo.YES.getValue()));
+        //then
+        HearingWindow hearingWindow = SscsCaseDataUtils.getHearingWindow(sscsCaseData);
+        assertEquals("2022-02-26", hearingWindow.getHearingWindowDateRange().getHearingWindowStartDateRange());
+        assertNull(hearingWindow.getHearingWindowDateRange().getHearingWindowEndDateRange());
+    }
+
+    @Test
+    void shouldGetHearingWindowWhenNonUrgentCase() {
+        // given
+        SscsCaseData sscsCaseData = Mockito.mock(SscsCaseData.class);
+        // when
+        Mockito.when(sscsCaseData.getEvents()).thenReturn(getEventsOfCaseData());
+        Mockito.when(sscsCaseData.getUrgentCase()).thenReturn((YesNo.NO.getValue()));
+        //then
+        HearingWindow hearingWindow = SscsCaseDataUtils.getHearingWindow(sscsCaseData);
+        assertEquals("2022-03-12", hearingWindow.getHearingWindowDateRange().getHearingWindowStartDateRange());
+        assertNull(hearingWindow.getHearingWindowDateRange().getHearingWindowEndDateRange());
+    }
+
+    @Test
+    void shouldGetNoHearingWindowWhenCaseResponseOverdue() {
+        // given
+        SscsCaseData sscsCaseData = Mockito.mock(SscsCaseData.class);
+        // when
+        Mockito.when(sscsCaseData.getEvents()).thenReturn(getOverdueEventsOfCaseData());
+        Mockito.when(sscsCaseData.getUrgentCase()).thenReturn((YesNo.YES.getValue()));
+        //then
+        HearingWindow hearingWindow = SscsCaseDataUtils.getHearingWindow(sscsCaseData);
+        assertNull(hearingWindow.getHearingWindowDateRange().getHearingWindowStartDateRange());
+        assertNull(hearingWindow.getHearingWindowDateRange().getHearingWindowEndDateRange());
+    }
+
+    @Test
+    void shouldGetCaseFlags() {
+        // given
+        SscsCaseData sscsCaseData = Mockito.mock(SscsCaseData.class);
+        MockedStatic<PartyFlagsMapping> partyFlagsMapping = Mockito.mockStatic(PartyFlagsMapping.class);
+        // when
+        partyFlagsMapping.when(() -> PartyFlagsMapping.getPartyFlags(sscsCaseData)).thenReturn(getPartyFlags());
+        // then
+        CaseFlags caseFlags = SscsCaseDataUtils.getCaseFlags(sscsCaseData);
+        assertNull(caseFlags.getFlagAmendUrl());
+        assertEquals(8, caseFlags.getFlags().size());
+        assertEquals(SIGN_LANGUAGE_TYPE.getFlagId(), caseFlags.getFlags().stream().findFirst().orElseThrow().getFlagId());
+    }
+
+    private List<ExcludeDate> getExcludeDates() {
```

Comment:
minor/optional using the List.of() api instead of arrayList would make these methods a lot cleaner

Reply:
Deleted.